### PR TITLE
Homework submission by Nate Taylor

### PR DESCRIPTION
**Pull request by nhtaylor**

### CODE REVIEW

File: pset-functions.js [hunk 1]
```diff
@@ -30,6 +30,35 @@ Use the following test cases to confirm your program meets the success criteria
...
+  } else if (playerCardScore > 21 && dealerCardScore > 21) {
+    // if both players bust
+    return 0
+  } else if (playerCardScore > 21 && dealerCardScore <= 21) {
+    // if player busts
+    return dealerCardScore
+  } else if (playerCardScore <= 21 && dealerCardScore > 21) {
+    // if dealer busts
+    return playerCardScore
+  } else if (playerCardScore > dealerCardScore) {
```
Comment (RaulJiminian) [inline pset-functions.js:47]: Nate, really great use of if/else statements here; I love that you considered an edge case where a user may not input a number as your first filter. The way you set up your if/else cascades perfectly for all potential scenarios in the proper order. Once you get here to line 47, you can also take advantage of built in Math methods that could help you avoid running additional if/else blocks (in this specific example). For a sample alternate solution using Math methods please check [here](https://github.com/jsd20200127/hw-02-functions-solution/blob/e87c29073cf7285a5045ec8708561f279474f2d2/pset-functions.js#L40-L41)

File: pset-functions.js [hunk 2]
```diff
@@ -75,6 +104,44 @@ Use the following test cases to confirm your program meets the success criteria
...
+      score += 8
+    } else if (word[i] === 'k') {
+      score += 5
+    } else if (word[i] === 'f' || word[i] === 'h' || word[i] === 'v' || word[i] === 'w' || word[i] === 'y') {
+      score += 4
+    } else if (word[i] === 'b' || word[i] === 'c' || word[i] === 'm' || word[i] === 'p') {
+      score += 3
+    } else if (word[i] === 'd' || word[i] === 'g') {
+      score += 2
+    } else if (word[i] === 'a' || word[i] === 'e' || word[i] === 'i' || word[i] === 'o' || word[i] === 'u' || word[i] === 'l' || word[i] === 'n' || word[i] === 'r' || word[i] === 's' || word[i] === 't') {
```
Comment (RaulJiminian) [inline pset-functions.js:134]: Nate, another great use of if/else statements in combination with the "or" operator to test your conditions. It's clear you are also keeping edge cases in mind (very well done). Another potential solution to using several "or" operands takes advantage of JS' built in method, .includes. For an example of this, please take a look [here](https://github.com/jsd20200127/hw-02-functions-solution/blob/e87c29073cf7285a5045ec8708561f279474f2d2/pset-functions.js#L98)

File: pset-functions.js [hunk 3]
```diff
@@ -114,6 +181,41 @@ Use the following test cases to confirm your program meets the success criteria
...
+  // making sure we're dealing with a string
+  if (typeof word === "string") {
+    word = word.toLowerCase()
+  } else {
+    return "isPalindrome() requires a string parameter."
+  }
+
+  // palindrome logic
+  for (i = 0; i < (word.length / 2); i++) {
+    if (word[i] === word[word.length - (1 + i)]) {
```
Comment (RaulJiminian) [inline pset-functions.js:199]: Nate, brilliant solution! You are checking both ends and meeting in the middle (without using built in methods)! If you did want to make use of some of javascript's built in methods, please find an alternate solution [here](https://github.com/jsd20200127/hw-02-functions-solution/blob/e87c29073cf7285a5045ec8708561f279474f2d2/pset-functions.js#L163)

File: pset-functions.js [hunk 4]
```diff
@@ -147,6 +249,39 @@ Use the following test cases to confirm your program meets the success criteria
...
+    word = word.toLowerCase()
+  } else {
+    return "doubleLetters() requires a string parameter."
+  }
+
+  // double adjacent letter logic
+  for (i = 0; i < (word.length - 1); i++) {
+    if (word[i] === word[1 + i]) {
+      result = true
+    } else {
```
Comment (RaulJiminian) [inline pset-functions.js:268]: Nate, you're crushing these examples; you are simultaneously comparing the array at different indices and taking advantage of where you are in the loop's iteration, really nice job! Small clean up, you can get rid of this else block if it's not doing anything. An else statement is not always required.

File: pset-functions.js [hunk 5]
```diff
@@ -208,6 +343,41 @@ console.log('Problem 5 - bonus:')
...
+  for (let i = 0; i < wordArray.length; i++) {
+    word = wordArray[i]
+    if (countArray[word] === undefined) {
+      countArray[word] = 1
+    } else {
+      countArray[word] += 1
+    }
+  }
+  return countArray
+}
```
Comment (RaulJiminian) [inline pset-functions.js:374]: Nate, I love how you used an object to store your updated word count; great use of a for loop mixed with conditional if/else statements. I notice you are removing punctation with is another great example of taking potential edge cases into consideration.

Overall, you did a really amazing job on this homework! All edge cases considered, very clean formatted code, and great logic to finding solutions. Keep up the great work!
